wallet-web/components/bond/BondNodeForm.tsx

```diff
@@ -120,59 +134,49 @@ export default function BondNodeForm(props: TBondNodeFormProps) {
     identityKey,
     host,
     version,
-    verlocPort,
     location,
-    mixPort,
-    httpApiPort,
-    clientsPort,
-  }: TFormInput): boolean => {
-    let validAmount = validateAmount(amount)
-    let validSphinxKey = validateKey(sphinxKey)
-    let validIdentityKey = validateKey(identityKey)
-    let validHost = validateHost(host)
-    let validVersion = validateVersion(version)
-
-    let validLocation =
-      props.type == NodeType.Gateway ? validateLocation(location) : true
-
+    verlocPort = DEFAULT_VERLOC_PORT.toString(),
+    mixPort = DEFAULT_MIX_PORT.toString(),
+    httpApiPort = DEFAULT_HTTP_API_PORT.toString(),
+    clientsPort = DEFAULT_CLIENTS_PORT.toString(),
+  }: TFormData): boolean => {
+    console.log({
+      amount,
+      sphinxKey,
+      identityKey,
+      host,
+      version,
+      verlocPort,
+      location,
+      mixPort,
+      httpApiPort,
+      clientsPort,
+    })
     let newValidity = {
-      validAmount: validAmount,
-      validSphinxKey: validSphinxKey,
-      validIdentityKey: validIdentityKey,
-      validHost: validHost,
-      validVersion: validVersion,
-      validLocation: validLocation,
-    }
-
-    if (advancedShown) {
-      let validMixPort = validateRawPort(mixPort)
-      let validVerlocPort =
-        props.type == NodeType.Mixnode ? validateRawPort(verlocPort) : true
-      let validHttpApiPort =
-        props.type == NodeType.Mixnode ? validateRawPort(httpApiPort) : true
-      let validClientsPort =
-        props.type == NodeType.Gateway ? validateRawPort(clientsPort) : true
-
-      newValidity = {
-        ...newValidity,
-        ...{
-          validMixPort: validMixPort,
-          validVerlocPort: validVerlocPort,
-          validHttpApiPort: validHttpApiPort,
-          validClientsPort: validClientsPort,
-        },
-      }
+      validAmount: validateAmount(amount),
+      validSphinxKey: validateKey(sphinxKey),
+      validIdentityKey: validateKey(identityKey),
+      validHost: validateHost(host),
+      validVersion: validateVersion(version),
+      validMixPort: validateRawPort(mixPort),
+      validLocation:
+        props.type == NodeType.Gateway ? validateLocation(location) : true,
+      validVerlocPort:
+        props.type === NodeType.Mixnode ? validateRawPort(verlocPort) : true,
+      validHttpApiPort:
+        props.type === NodeType.Mixnode ? validateRawPort(httpApiPort) : true,
+      validClientsPort:
+        props.type === NodeType.Gateway ? validateRawPort(clientsPort) : true,
```

Comment:
I'm just validating everything here including the values in the advanced area. If the advanced area is show then we validate the values that are supplied by the user. If the advanced area is not shown then we are using the defaults from above which will be valid.

Reply:
I guess this would simplify stuff